test/tests/p5.SoundRecorder.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+define(['chai'], function(chai) {
+  var expect = chai.expect;
+
+  describe('p5.SoundRecorder', function() {
+    var recorder;
+    var inputSoundFile;
+    var originalWriteFile;
+    var writeFileCalled;
+
+    before(function(done) {
+      // clean up audio nodes from other test suites
+      p5.soundOut.soundArray.length = 0;
```

Comment:
nice, you found the soundArray!

Setting length to 0 isn't something I've seen before. Will this free up the resources for garbage collection?

An approach we've taken is to explicitly call `dispose` on the nodes in this array. But if we are properly disposing at the end of each test, this may not be necessary?

Reply:
I think it's better to assume that other tests are properly disposing of their resources, so I've taken this line out entirely.

Reply:
👍